ibm/service/secretsmanager/resource_ibm_sm_username_password_secret.go

```diff
@@ -0,0 +1,487 @@
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"description": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "An extended description of your secret.To protect your privacy, do not use personal data, such as your name or location, as a description for your secret group.",
+			},
+			"expiration_date": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The date a secret is expired. The date format follows RFC 3339.",
+			},
+			"labels": &schema.Schema{
+				Type:        schema.TypeList,
+				Optional:    true,
+				Description: "Labels that you can use to search for secrets in your instance.Up to 30 labels can be created.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"name": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "A human-readable name to assign to your secret.To protect your privacy, do not use personal data, such as your name or location, as a name for your secret.",
+			},
+			"secret_group_id": &schema.Schema{
+				Type:        schema.TypeString,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "A v4 UUID identifier, or `default` secret group.",
+			},
+			"secret_type": &schema.Schema{
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The secret type. Supported types are arbitrary, certificates (imported, public, and private), IAM credentials, key-value, and user credentials.",
+			},
+			"version_custom_metadata": &schema.Schema{
+				Type:        schema.TypeMap,
+				Optional:    true,
+				ForceNew:    true,
+				Description: "The secret version metadata that a user can customize.",
+				Elem:        &schema.Schema{Type: schema.TypeString},
+			},
+			"rotation": &schema.Schema{
+				Type:        schema.TypeList,
+				MaxItems:    1,
+				Optional:    true,
+				Computed:    true,
+				Description: "Determines whether Secrets Manager rotates your secrets automatically.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"auto_rotate": &schema.Schema{
+							Type:        schema.TypeBool,
+							Optional:    true,
+							Description: "Determines whether Secrets Manager rotates your secret automatically.Default is `false`. If `auto_rotate` is set to `true` the service rotates your secret based on the defined interval.",
+						},
+						"interval": &schema.Schema{
+							Type:        schema.TypeInt,
+							Optional:    true,
+							Computed:    true,
+							Description: "The length of the secret rotation time interval.",
+						},
+						"unit": &schema.Schema{
+							Type:        schema.TypeString,
+							Optional:    true,
+							Computed:    true,
+							Description: "The units for the secret rotation time interval.",
+						},
+					},
+				},
+			},
+			"username": &schema.Schema{
```

Comment:
username and password are optional attributes if this resource is to create username and password secret type

Reply:
Done

Reply:
But still I see them has optional username and password>
Do we need mark them as Required?